compiler/compiler_test.go

```diff
@@ -475,6 +475,52 @@ func TestCompileAndEval(t *testing.T) {
 			}),
 			want: values.NewBool(true),
 		},
+		{
+			name: "null array",
+			fn:   `(r) => r.a[0]`,
+			inType: semantic.NewObjectType([]semantic.PropertyType{
+				{Key: []byte("r"), Value: semantic.NewObjectType([]semantic.PropertyType{
+					{Key: []byte("a"), Value: semantic.NewArrayType(semantic.BasicString)},
+				})},
+			}),
+			input: values.NewObjectWithValues(map[string]values.Value{
+				"r": values.NewObjectWithValues(nil),
+			}),
+			wantEvalErr: true,
+		},
+		{
+			name: "null record",
+			fn:   `(r) => r.a["b"]`,
+			inType: semantic.NewObjectType([]semantic.PropertyType{
+				{Key: []byte("r"), Value: semantic.NewObjectType([]semantic.PropertyType{
+					{Key: []byte("a"), Value: semantic.NewObjectType([]semantic.PropertyType{
+						{Key: []byte("b"), Value: semantic.BasicString},
+					})},
+				})},
+			}),
+			input: values.NewObjectWithValues(map[string]values.Value{
+				"r": values.NewObjectWithValues(nil),
+			}),
+			wantEvalErr: true,
+		},
+		// TODO(jsternberg): We presently have not implemented dictionary support for
+		// runtime functions. There aren't any builtins that use this functionality,
+		// but when we do, this test will need to be uncommented to ensure that
+		// a null dictionary does not sneak in.
```

Comment:
I'm usually hesitant to leave commented code in, but this is quite helpful. Do we have a timeframe from when we'll support this? Is there an issue we can add a comment to about uncommenting this?

Reply:
I'm not sure honestly. I wish I had a better answer than that.